### ISSUE TITLE
PLW1641: false positive for classes inheriting from builtins

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#10025]
```python
class Foo(dict):
    def __eq__(self, other):
        return # some logic
```
reports `PLW1641 eq-without-hash` but `__hash__` is inherited from `dict`.

**Comment by trag1c** [comment]
Isn't `dict.__hash__` `None`?

**Comment by trag1c** [comment]
Also:
> A class that overrides [__eq__()](https://docs.python.org/3/reference/datamodel.html#object.__eq__) and does not define [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) will have its [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) implicitly set to None.

Source: <https://docs.python.org/3/reference/datamodel.html#object.__hash__>

**Comment by spaceone** [comment]
oh yes, sorry.
